#589 Implement firebase app check

HalcyonJAC opened the pull request:
## What's included?
Here we are adding the recaptcha element to the vue app in order to enable the Firebase App Check:
https://firebase.google.com/docs/app-check

## Who should test?
✅ Developers

## How to test?
The App Check requires a key which I have stored in an env variable.

Someone will need to create one for the live platform, but I will share the one I created in slack.

### Step 1 (this PR)
Add the app check to the front end application and ensure it is being correctly passed in when firebase is requested (realtime db and storage). To check this, you can go to: https://console.firebase.google.com/u/4/project/digital-platform-develop/settings/appcheck and click Storage or Realtime Database to see the % of authorised requests.

### Step 2 (requires a new ticket)
Enforce it across the application. This will mean:

- Clicking the Enforce buttons across the various platforms
- Adding a check to all callable functions to check for `context.app` and reject the request if this not present (https://firebase.google.com/docs/app-check/cloud-functions)

### Note: 
This will prevent tools like postman from working when making requests to firebase.

## Risk - how likely is this to impact other areas?
🔴 High risk - this includes a lot of changes to shared code

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_

warrensearle reviewed (changes requested):
Changes here look good, just need to get the new `VUE_APP_RECAPTCHA_TOKEN` environment variable in the github workflow files and, I think, the `.env` files.
Let me know if any questions around the github setup

HalcyonJAC commented:
Thanks @warrensearle I've updated the env files, workflow files and added the secret in github for **develop only**. Please take a look and let me know if it's okay!

warrensearle reviewed (approved):
Thanks @HalcyonJAC looks good 👍
